src/core/meshcop/border_agent.cpp

```diff
@@ -218,57 +156,68 @@ void BorderAgent::HandleCoapResponse(void *               aContext,
 {
     OT_UNUSED_VARIABLE(aMessageInfo);
 
-    ForwardContext &     forwardContext = *static_cast<ForwardContext *>(aContext);
-    BorderAgent &        borderAgent    = forwardContext.GetBorderAgent();
-    Instance &           instance       = borderAgent.GetInstance();
-    const Coap::Message *response       = static_cast<const Coap::Message *>(aMessage);
-    Coap::Message *      message        = nullptr;
-    otError              error;
+    ForwardContext &forwardContext = *static_cast<ForwardContext *>(aContext);
+
+    forwardContext.Get<BorderAgent>().HandleCoapResponse(forwardContext, static_cast<const Coap::Message *>(aMessage),
+                                                         aResult);
+}
+
+void BorderAgent::HandleCoapResponse(ForwardContext &aForwardContext, const Coap::Message *aResponse, otError aResult)
```

Comment:
Can `aResponse` be set to `nullptr`? If not, pass by reference?

Reply:
Was thinking of the same, but checked the callback and its use, and it can be set to  `nullptr` in case of error (e.g. `aResult == OT_ERROR_RESPONSE_TIMEOUT`)  when no response is received.